pkg/volume/csi/csi_attacher.go

```diff
@@ -235,18 +271,34 @@ func (c *csiAttacher) VolumesAreAttached(specs []*volume.Spec, nodeName types.No
 	attached := make(map[*volume.Spec]bool)
 
 	for _, spec := range specs {
+		var driver, handle string
 		if spec == nil {
 			klog.Error(log("attacher.VolumesAreAttached missing volume.Spec"))
 			return nil, errors.New("missing spec")
 		}
-		source, err := getCSISourceFromSpec(spec)
+
+		volSrc, pvSrc, err := getSourceFromSpec(spec)
 		if err != nil {
 			klog.Error(log("attacher.VolumesAreAttached failed: %v", err))
 			continue
 		}
-		skip, err := c.plugin.skipAttach(source.Driver)
+
+		if c.plugin.inlineFeatureEnabled && volSrc != nil {
+			driver = volSrc.Driver
+			if volSrc.VolumeHandle != nil {
+				handle = *volSrc.VolumeHandle
+			}
```

Comment:
its ok if `VolumeHandle` is nil here? I feel like this should be an error. Another reason why this block should be factored out with the other ones that do the same thing, more consistent error handling.